### ISSUE TITLE
[Bug] Account-Expires attribute produces invalid UNIX timestamp

### ISSUE THREAD

**Issue by striker4150** [DirectoryTree/LdapRecord#552]
**Environment:**
 - LDAP Server Type: ActiveDirectory
 - PHP Version: 8.1

**Describe the bug:**
I encountered a TypeError on line [231](https://github.com/DirectoryTree/LdapRecord/blob/master/src/Models/Attributes/Timestamp.php#L231) in `src/Models/Attributes/Timestamp.php`. `DateTime::setTimestamp` expects an `int`, but if you look at the typing of [Utilities::convertWindowsTimeToUnixTime](https://github.com/DirectoryTree/LdapRecord/blob/master/src/Utilities.php#L156), it returns a `float`.

After a bit of investigating, I found that this is due to the [accountexpires](https://learn.microsoft.com/en-us/windows/win32/adschema/a-accountexpires) AD attribute. If the user account is set to never expire, it can be set to the value 9223372036854775807, which cannot be converted to a valid UNIX timestamp. In fact, the value represents Sep 13, 30828, which is way outside of the valid UNIX timestamp range. It might be worthwhile to limit the values passed to the `Utilities::convertWindowsTimeToUnixTime` function to between 116444736000000000 (the minimum convertible Windows timestamp) and 137919572470000000 (the maximum convertible Windows timestamp), but I'm not sure how you would like to handle this.

I don't know whether any other attributes have similar sizing issues, but fixing this should at least fix any timestamp attributes.

**Comment by stevebauman** [comment]
Hi @striker4150, thanks for the report!

I'll have this patched this weekend 🙏 , though I of course welcome any PR's if you'd like this resolved sooner.

**Comment by stevebauman** [comment]
Hi @striker4150!

I've patched this issue in #553 and I will have a new release out momentarily.

Please run `composer update` to get the latest version. Then, once updated, here is how this is handled:

If you have an account control value that's set to the maximum (`9223372036854775807`) or minimum (`0`) windows integer time values, then those values will be returned when retrieving them from the `User` model.

You can use these values to then determine if the account never expires.

For example:

```php
use LdapRecord\Models\Attributes\Timestamp;

$user = User::find('cn=jdoe,dc=local,dc=com');

if (in_array($user->accountExpires, [0, Timestamp::WINDOWS_INT_MAX], $strict = true) {
    // The user account never expires.
}
```

I'm not sure if it's worth implementing the other circumstances at this time, as it would rare for someone to set dates past the maximum's you've sent.

If you think otherwise, let me know and we can figure out a solution there as well 🙏
